manifests/init.pp

```diff
@@ -36,6 +36,77 @@
   $vcl_conf             = '/etc/varnish/default.vcl',
 ) {
 
+  # validate parameters
+  validate_bool($manage_default_vcl)
+
+  validate_absolute_path($vcl_path)
+
+  if is_ip_address($admin_listen_address) == false {
+    fail('must be a valid IP address')
+  }
+
+  validate_re("${admin_listen_port}", '^\d+$', 'is not a valid server port') # lint:ignore:only_variable_string
+  if $admin_listen_port == 0 {
+    fail('is not a valid server port')
```

Comment:
we should list the actual value, so this makes more sense to the user. It will still work with your current regex in the spec test.

```puppet
fail("varnish::admin_list_port is not a valid server port. Detected value is <${admin_list_port}>")
```